oauth2/src/main/java/com/okta/spring/boot/oauth/env/OIDCMetadata.java

```diff
@@ -0,0 +1,85 @@
+package com.okta.spring.boot.oauth.env;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.http.ResponseEntity;
+
+public class OIDCMetadata {
+
+    private boolean isAuth0;
+    private final String clientAuthenticationMethod = "none";
+    private final String scope = "profile,email,openid";
+    private final String jwkSetURI;
+    private final String authorizationURI;
+    private final String tokenURI;
+    private final String userInfoURI;
+    private final String introspectionURI;
+
+    public boolean isAuth0() {
+        return isAuth0;
+    }
+
+    public String getClientAuthenticationMethod() {
+        return clientAuthenticationMethod;
+    }
+
+    public String getScope() {
+        return scope;
+    }
+
+    public String getJwkSetURI() {
+        return jwkSetURI;
+    }
+
+    public String getAuthorizationURI() {
+        return authorizationURI;
+    }
+
+    public String getTokenURI() {
+        return tokenURI;
+    }
+
+    public String getUserInfoURI() {
+        return userInfoURI;
+    }
+
+    public String getIntrospectionURI() {
+        return introspectionURI;
+    }
+
+    public OIDCMetadata(String issuerWithPathKey) {
+        this.jwkSetURI = "${" + issuerWithPathKey + "}/v1/keys";
+        this.authorizationURI = "${" + issuerWithPathKey + "}/v1/authorize";
+        this.tokenURI = "${" + issuerWithPathKey + "}/v1/token";
+        this.userInfoURI = "${" + issuerWithPathKey + "}/v1/userinfo";
+        this.introspectionURI = "${" + issuerWithPathKey + "}/v1/introspect";
+    }
+
+    /**
+     * Fetch metadata from the ${issuer}/.well-known/openid-configuration endpoint
+     *
+     * @param response well known metadata response
+     * @param issuer   OIDC issuer URI
+     */
+    public OIDCMetadata(ResponseEntity<String> response, String issuer) throws JsonProcessingException {
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode root = mapper.readTree(response.getBody());
+
+        // TODO check if there are other possible domains
+        if (issuer != null && issuer.contains("auth0.com")) {
+            this.isAuth0 = true;
+            // TODO temp workaround as auth0 does not have this URL
+            this.introspectionURI = issuer + "introspect";
```

Comment:
@arvindkrishnakumar-okta @mraible this is my main concern. Auth0 does not support introspection URL so we need to find a workaround for this. Any suggestions?

Reply:
@deepu105 since Auth0 does not have introspection URL in OIDC metadata, we can simply ignore/NOT set this variable? Does that work?

Reply:
If I dont set the URL then spring still tries to configure a bean for introspection and fails. Which is not that case if we setup spring security without using okta starter, so I assume something in the code is making SPring to autoconfigure it. I might need help figuring this out

Reply:
this is fixed by not using opaque mode for auth0 domains